### PR TITLE
SNOW-1349069: Update cache_result comments in apply methods

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1349069

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Link to SNOW-1345395 and explain that we need to keep the `cache_result` calls.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -2873,6 +2873,11 @@ def groupby_apply(
         |        1 |        2 | k1                   |                 14 | b                     |                  1 |
         |        0 |        0 | k0                   |                 15 | c                     |                  2 |
         """
+        # NOTE we are keeping the cache_result for performance reasons. DO NOT
+        # REMOVE the cache_result unless you can prove that doing so will not
+        # materially slow down CI or individual groupby.apply() calls.
+        # TODO(SNOW-1345395): Investigate why and to what extend the cache_result
```
Comment (sfc-gh-lmukhopadhyay) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:2879]: nit: should be extent?

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:2879]:
```suggestion
        # TODO(SNOW-1345395): Investigate why and to what extent the cache_result
```

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 2]
```diff
@@ -5612,8 +5617,11 @@ def _apply_with_udtf_and_dynamic_pivot_along_axis_1(
             ).over(partition_by=[partition_identifier]),
         )
 
-        # TODO SNOW-1060191: after fixing the bug in PIVOT with udtf, remove cache_result
-        # cache_result is currently used to create a temp table and read from it
+        # NOTE we are keeping the cache_result for performance reasons. DO NOT
+        # REMOVE the cache_result unless you can prove that doing so will not
+        # materially slow down CI or individual groupby.apply() calls.
+        # TODO(SNOW-1345395): Investigate why and to what extend the cache_result
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:5623]:
```suggestion
        # TODO(SNOW-1345395): Investigate why and to what extent the cache_result
```

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM.

**Pull request merged**